projects/angular/src/utils/loading/loading.ts

```diff
@@ -40,11 +41,18 @@ export class ClrLoading implements OnDestroy {
     }
 
     this._loadingState = value;
-    if (this.listener) {
+    if (this.listener && this._viewInitialized) {
       this.listener.loadingStateChange(value);
     }
   }
 
+  ngAfterViewInit() {
+    this._viewInitialized = true;
+    setTimeout(() => {
```

Comment:
This will trigger change detections. Apps can have many ClrLoading instances within itself. So we have to avoid solutions that rely on timeouts.

Reply:
We avoid the `setTimeout` workaround in almost all cases. We only use it when we exhaust all other options and its use cases would be limited within the app.

Reply:
I have another solution that I proposed up in the comments (using visibility:hidden), but I thought this was the lesser of the two evils. I could avoid the `setTimeout` by forcing change detection with `ChangeDetectorRef.detectChanges`. That will still trigger another change detection, but at least it will be more explicit. Perhaps there's a better way, but my strategy was to render the button to get the proper size and then immediately hide it.